### PR TITLE
Use travis to generate MacOS binary releases

### PR DESCRIPTION
**Pull request by lorenzo**
The use of travis stages is so that we can stay in the allowed time limits for travis. Removed the testing part out of travis, since we have plenty tests already, but it can be added back with no problem.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -24,37 +20,103 @@ cache:
...
+       env: GHC_VER="8.4.2"
+       script: *test
+
+     - stage: test
+       env: GHC_VER="8.2.2"
+       script: *test
+
+     - stage: test
+       env: GHC_VER="8.2.1"
+       script: *test
```
Comment (bubba) [inline .travis.yml]: Should we also add 8.6.1/8.6.2

Comment (lorenzo) [reply .travis.yml]: Yeah, but I was waiting for your recent PR to be merged first.

**Comment by alanz** [comment]
When you guys are happy, merge.

I presume we can get linux binaries out of this as well?

**Comment by alanz** [comment]
And in terms of S3 or other, if we ask the haskell committee they may be able to help

**Comment by lorenzo** [comment]
I think this is good to merge already. Maybe circleci is good for Linux deployments?

**Comment by lorenzo** [comment]
yeah, I’ll start experimenting with the windows builds next. Likely in my own fork to not disturb circleci and appveyor

**Comment by lorenzo** [comment]
Ok, merging this. I started to work on windows builds

**Pull request merged**